obspy/core/util/base.py

```diff
@@ -646,6 +647,9 @@ def download_to_file(url, filename_or_buffer, chunk_size=1024):
 
 def _generic_reader(pathname_or_url=None, callback_func=None,
                     **kwargs):
+    # convert pathlib.Path objects to str for compatibility.
+    if isinstance(pathname_or_url, PurePath):
+        pathname_or_url = str(Path(pathname_or_url))
```

Comment:
No need to cast to `Path`, `__str__` is implemented by `PurePath` as parent anyway

Reply:
The reason I included this is a bit esoteric, but here is the weird edge case I had in mind:
```python
import pathlib
# assume we are using linux
windows_path = pathlib.PureWindowsPath('base') / 'filename.mseed'
print(str(windows_path))  # prints 'path\\filename.mseed' which is not a valid path for linux, hence we need the extra Path call
```
I don't think this will really happen much, if at all, but the code as it is written will handle it.

Reply:
I still don't get it.. for me using..

```
print(str(pathlib.Path(windows_path)))
```

..in there doesn't change anything on the output

Reply:
Ok this is probably too far of an edge case anyway (why would someone use a PureWindowsPath on linux with ObSpy to read a file?) so I am going to just remove the extra `Path` constructor.